### PR TITLE
Parametrization "projection of boundary vectors" approach

### PR DESCRIPTION
**Pull request by ksstar**
- Created a new assign function: AssignZeroDirichlet --> Zero internal field + all boundary conditions set to homogenous Dirichlet.
- Modified reduced matrices: RC_matrix and RD_matrix --> Looping over all parametric boundary conditions. The boundary condition on the patch of interest is set to unity. A matrix for each parametric boundary condition is stored.
- Modified reduced matrices: LinSysDiv,  LinSysConv, LinSysDiff-->Looping over all parametric boundary conditions. 
- Modified ReducedUnsteadyNSExplicit such that the value of the velocity boundary condition can be imposed at the reduced order level.
- Added Consistent Flux Method to ReducedUnsteadyNSExplicit.

- Updated CFD tutorial 17: fluid enters the two inlets of the Yjunction under an angle of 60 degrees (was 45 degrees)
- Modified the M_Assert of reductionProblem::computeLift such that the function is also valid for negative values
- Updated the iterative penalty method (reducedUnsteadyNS::penalty_sup and reducedUnsteadyNS::penalty_PPE) for second order time derivatives (timeDerivativeSchemeOrder)

- Added documentation tutorial 17 and 19.

### CODE REVIEW

File: src/ITHACA_FOMPROBLEMS/steadyNS/steadyNS.C [hunk 1]
```diff
@@ -353,6 +353,7 @@ void steadyNS::projectPPE(fileName folder, label NU, label NP, label NSUP)
     NPmodes = NP;
     NSUPmodes = 0;
     L_U_SUPmodes.resize(0);
+    Vector<double> inl(0, 0, 0);
```
Comment (ksstar) [inline src/ITHACA_FOMPROBLEMS/steadyNS/steadyNS.C:356]: To be removed

File: src/ITHACA_FOMPROBLEMS/steadyNS/steadyNS.C [hunk 2]
```diff
@@ -543,6 +544,7 @@ void steadyNS::projectSUP(fileName folder, label NU, label NP, label NSUP)
     NPmodes = NP;
     NSUPmodes = NSUP;
     L_U_SUPmodes.resize(0);
+    Vector<double> inl(0, 0, 0);
```
Comment (ksstar) [inline src/ITHACA_FOMPROBLEMS/steadyNS/steadyNS.C:547]: To be removed

File: src/ITHACA_FOMPROBLEMS/steadyNS/steadyNS.H [hunk 3]
```diff
@@ -142,6 +142,7 @@ class steadyNS: public reductionProblem
         /// Number of nut modes used for the projection
         label NNutModes;
 
+
```
Comment (ksstar) [inline src/ITHACA_FOMPROBLEMS/steadyNS/steadyNS.H:145]: To be removed

File: tutorials/CFD/17YJunction/17YJunction.C [hunk 4]
```diff
@@ -335,7 +341,8 @@ int main(int argc, char* argv[])
     std::cout << "elapsed_ROM: " << elapsed_ROM.count() << " seconds.";
     std::cout << std::endl;
     auto start_ROM_REC = std::chrono::high_resolution_clock::now();
-    reduced.reconstruct_PPE("./ITHACAoutput/ReconstructionPPE");
+    //reduced.reconstruct_PPE("./ITHACAoutput/ReconstructionPPE");
```
Comment (ksstar) [inline tutorials/CFD/17YJunction/17YJunction.C:344]: To be removed

File: src/ITHACA_CORE/ITHACAutilities/ITHACAassign.H [hunk 5]
```diff
@@ -253,6 +253,22 @@ template<typename Type>
...
+//--------------------------------------------------------------------------
+/// Assign zero internal field.
+/// Change all non-zero Dirichlet boundary condition values to zero.
+/// Change all Neumann boundary conditions to Dirichlet boundary conditions.
+///
+/// @param[in,out]  field  Field of which the BCs need to be changed
+///
+/// @tparam         Type   Type of field (vector or scalar)
+///
+void assignZeroDirichlet(GeometricField<vector, fvPatchField,
```
Comment (giovastabile) [inline src/ITHACA_CORE/ITHACAutilities/ITHACAassign.H]: we should use templates here

**Review by giovastabile — COMMENTED** [review]
Everything looks ok, only one minor thing

**Pull request merged**